sys/vm/vm_mmap.c

```diff
@@ -1497,6 +1497,17 @@ kern_mincore(struct thread *td, uintptr_t addr0, size_t len,
 				    pmap_is_referenced(m) ||
 				    (m->a.flags & PGA_REFERENCED) != 0)
 					mincoreinfo |= MINCORE_REFERENCED_OTHER;
+
+				/* Expose capability tracking information */
+				mincoreinfo |=
+					((m->oflags & VPO_CAPSTORE) != 0)
+						? MINCORE_CAPSTORE : 0;
+
+				mincoreinfo |=
+					((m->a.flags & PGA_CAPDIRTY) != 0)
+						? (MINCORE_CAPSTORE |
+						   MINCORE_CAPDIRTY)
```

Comment:
I feel like you only want `MINCORE_CAPDIRTY` here? Either you should assert that `PGA_CAPDIRTY => VPO_CAPSTORE` or you should not override it if that's a legitimate combo (say, a page is capdirty and then gets re-mapped read-only).